torch_xla/csrc/tensor_methods.cpp

```diff
@@ -2528,14 +2528,26 @@ XLATensorPtr XLATensor::trace(const XLATensorPtr& input) {
 XLATensorPtr XLATensor::transpose(const XLATensorPtr& input, int64_t dim0,
                                   int64_t dim1) {
   auto input_shape = input->shape();
-  auto permute_dims = torch::lazy::MakeTransposePermutation(
-      /*dim0=*/dim0, /*dim1=*/dim1, /*rank=*/input_shape.get().rank());
-  ViewInfo view_info(ViewInfo::Type::kPermute, input_shape, permute_dims);
+  ViewInfo view_info;
+  if (input_shape.get().rank() <= 1) {
+    // return a view of self if input rank <=1
```

Comment:
Since we want to make it a noop if the input tensor 1d, I wonder why we can't just return `input` here.

Reply:
oh I see. It's not a in-place operation so we need to create a copy of the tensor, such as the ViewInfo, right?